### PR TITLE
Expose and use bdev stat reset rpc

### PR DESCRIPTION
**Pull request by abhilashshetty04**

### CODE REVIEW

File: src/bdev_async.rs [hunk 1]
```diff
@@ -29,6 +30,14 @@ pub struct BdevAsyncCallContext {
     pub arg: *mut c_void,
 }
 
+/// Defines mode for Bdev IOStats reset.
+pub enum BdevResetMode {
+    /// Resets all IoStat counters for a given Bdev.
+    SpdkBdevResetStatAll = 0,
```
Comment (tiagolobocastro) [inline src/bdev_async.rs]: Can't we pull these from libspdk?

**Review by dsavitskiy — APPROVED** [review]

File: src/bdev_async.rs (hunk 1 above)
Comment (abhilashshetty04) [reply src/bdev_async.rs]: Done, We could import the enum variant and use.

File: src/bdev_async.rs [hunk 2]
```diff
@@ -94,8 +95,30 @@ where
         let errno = r.await.expect("Cancellation is not supported");
         errno_result_from_i32(stat, errno)
     }
+
+    /// This function Resets all stat counters for a given Bdev.
+    /// Returns Errno incase of an error.
```
Comment (tiagolobocastro) [inline src/bdev_async.rs]:
```suggestion
    /// This function resets all stat counters for a given Bdev.
    /// Returns Errno in case of an error.
```

**Review by tiagolobocastro — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors try

File: src/bdev_async.rs [hunk 3]
```diff
@@ -137,3 +159,23 @@ unsafe extern "C" fn inner_stats_callback(
...
+/// # Safety
+///
+unsafe extern "C" fn inner_stats_reset_callback(
+    _bdev: *mut spdk_bdev,
+    arg: *mut c_void,
+    errno: i32,
+) {
+    let s = Box::from_raw(arg as *mut oneshot::Sender<i32>);
+    s.send(errno)
+        .expect("`inner_stats_callback()` receiver is gone");
```
Comment (dsharma-dc) [inline src/bdev_async.rs]:
```suggestion
        .expect("`inner_stats_reset_callback()` receiver is gone");
```

**Review by dsharma-dc — APPROVED** [review]

**Pull request merged**